src/foam/dao/GUIDDAO.js

```diff
@@ -67,10 +67,9 @@ return (foam.core.PropertyInfo)(getOf().getAxiomByName(getProperty()));
       },
       javaCode: `
 Object val = obj.getProperty(getProperty());
-
+UIdGenerator uIdGenerator = (UIdGenerator) x.get("uIdGenerator");
```

Comment:
Do we want to share the id generator across DAO's or have each DAO have its own? I think per-DAO is sufficient.

Reply:
Per DAO would reduce any lock contention should there be any.